### DISCUSSION TITLE
Comparing a Rep[LocalDate] with a Rep[LocalDateTime]

### DISCUSSION THREAD

**Discussion by TheNthDev** [slick/slick#2733] [Questions] [answered]
How would I go about comparing a LocalDate and a LocalDateTime in Slick? Our application has a PostgreSQL database in which we have Date columns that are mapped to LocalDate objects and Timestamp columns that are mapped to LocalDateTime objects as of Slick 3.3 (we recently upgraded). How do I go about comparing the two columns in a Slick query? I've tried to use a custom mapping that will convert LocalDates to Timestamps:
`   /**`
`    * Used to compare a Rep[LocalDate] to a Rep[LocalDateTime]`
`    */`
`   val toLocalDateTimeFromLocalDate = SimpleExpression.unary[LocalDate, LocalDateTime] { (date, qb) =>`
`     qb.sqlBuilder += "CAST("`
`     qb.expr(date)`
`    qb.sqlBuilder += " as timestamp)"`
`   }`

While this solution works, it is not very efficient or intuitive. Furthermore, I believe that it is causing issues during run-time while generating some more complex queries that worked before migrating to 3.3.x. The only reason I believe that this is the issue is because I copied the raw query that Slick generates, and it will run fine directly against PostgreSQL, it just has issues in Slick. 

Edit: I've gotten `.asColumnOf[LocalDateTime]` to work, but it still has the same issue of doing the exact same SQL CAST, which can be inefficient.

**Comment by nafg** [comment, marked as answer]
If it should work in SQL without a cast then maybe for now the workaround would be .asInstanceOf[Rep[...]]

Why 3.3 and not 3.4?

**Reply by TheNthDev** [reply]
Thank you, I will try that and let you know.

As for the version of Slick, our application was using 3.2 and I updated it to 3.4 (technically 3.5.0-M3 just to test it, but I'll put it back down to 3.4). I just wanted to call out 3.3, since that was when the LocalDate/LocalDateTime mappers were added.

**Reply by TheNthDev** [reply]
Thank you, that worked splendidly. It didn't seem to help my strange run-time error, but it answered the specific question that I asked.
